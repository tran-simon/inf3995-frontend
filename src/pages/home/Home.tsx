import React from 'react';
import {
  Button,
  Container,
  Paper,
  TableContainer,
  Typography,
  useTheme,
} from '@material-ui/core';
import { FirebaseDatabaseMutation } from '@react-firebase/database';
import MapData from '../../model/MapData';
import MapsTable from './MapsTable';
import { useHistory } from 'react-router-dom';

/**
 * Home page
 *
 * Inspired from https://github.com/mui-org/material-ui/blob/master/docs/src/pages/getting-started/templates/album/Album.js
 *
 */
const Home = () => {
  const theme = useTheme();
  const history = useHistory();

  return (
    <>
      <main
        style={{
          display: 'flex',
          flexDirection: 'column',
          height: '100vh',
        }}
      >
        <Container
          maxWidth="sm"
          style={{
            padding: theme.spacing(8, 0, 6),
          }}
        >
          <Typography
            component="h1"
            variant="h2"
            align="center"
            gutterBottom
            style={{ fontSize: '3rem' }}
          >
            Centre de contrôle des Crazyflies
          </Typography>
          <Typography
            variant="h5"
            align="center"
            color="textSecondary"
            paragraph
          >
            INF3995 - Équipe 100
          </Typography>
        </Container>
        <Container maxWidth="md">
          <TableContainer
            component={Paper}
            style={{
              height: '50vh',
            }}
          >
            <MapsTable />
          </TableContainer>
          <FirebaseDatabaseMutation path={'/maps'} type={'push'}>
            {({ runMutation }) => (
              <Button
                variant="contained"
                style={{ width: '100%', margin: '1rem 0' }}
                onClick={() => {
                  runMutation(new MapData()).then((v) => {
                    v.key && history.push(`/explore/${v.key}`);
                  });
                }}
              >
                Nouvelle Exploration...
              </Button>
            )}
          </FirebaseDatabaseMutation>
        </Container>
      </main>
    </>
  );
};

export default Home;
