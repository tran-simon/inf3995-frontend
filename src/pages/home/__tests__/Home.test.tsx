import React from 'react';
import { fireEvent, render, waitFor, screen } from '@testing-library/react';
import Home, { NewExplorationBtn } from '../Home';
import { FirebaseDatabaseMutation } from '@react-firebase/database';

jest.mock('@react-firebase/database');

describe('Home', () => {
  it('can match snapshot', async () => {
    await waitFor(() => {
      expect(render(<Home />).asFragment()).toMatchSnapshot();
    });
  });

  it('opens new exploration popup on button click', async () => {
    render(<NewExplorationBtn runMutation={() => {}} />);
    expect(screen.queryByRole('dialog')).toBeNull();

    const btn = screen.getByRole('button');
    fireEvent.click(btn);

    await waitFor(() => {
      expect(screen.queryByRole('dialog')).not.toBeNull();
    });
  });
});
