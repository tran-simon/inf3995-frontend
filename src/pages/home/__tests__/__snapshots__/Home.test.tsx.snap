// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home can match snapshot 1`] = `
<DocumentFragment>
  <main
    style="display: flex; flex-direction: column; height: 100vh;"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthSm"
      style="padding: 64px 0px 48px;"
    >
      <h1
        class="MuiTypography-root MuiTypography-h2 MuiTypography-gutterBottom MuiTypography-alignCenter"
        style="font-size: 3rem;"
      >
        Centre de contrôle des Crazyflies
      </h1>
      <p
        class="MuiTypography-root MuiTypography-h5 MuiTypography-colorTextSecondary MuiTypography-paragraph MuiTypography-alignCenter"
      >
        INF3995 - Équipe 100
      </p>
    </div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthMd"
    >
      <div
        class="MuiPaper-root MuiTableContainer-root MuiPaper-elevation1 MuiPaper-rounded"
        style="height: 50vh;"
      />
    </div>
  </main>
</DocumentFragment>
`;
